### ISSUE TITLE
[Enhancement]: Slightly more flexible API for add_missing_bounds

### ISSUE THREAD

**Issue by pochedls** [xCDAT/xcdat#666]
### Is your feature request related to a problem?

This is a convenience request. I almost always need to add missing bounds for one axis, but the (updated?) API [wants a list](https://github.com/xCDAT/xcdat/blob/main/xcdat/bounds.py#L127). 

### Describe the solution you'd like

I'd like the API to accept a list **or string** (e.g., `["X", "Y"]` or simply `T`).

**Comment by lee1043** [comment]
I like this idea!

**Comment by tomvothecoder** [comment]
Sounds good to me. These APIs also need to be updated:

- https://github.com/xCDAT/xcdat/blob/c653bf2eae50bb86100b56611d9dc7ebfb4bd217/xcdat/dataset.py#L44-L47
- https://github.com/xCDAT/xcdat/blob/c653bf2eae50bb86100b56611d9dc7ebfb4bd217/xcdat/dataset.py#L127-L130
- https://github.com/xCDAT/xcdat/blob/c653bf2eae50bb86100b56611d9dc7ebfb4bd217/xcdat/dataset.py#L463-L467

**Comment by tomvothecoder** [comment]
> When opening a dataset it kind of makes sense to pass in a list for most use cases (make sure this has time/lat/lon bounds no matter what...). A lot of times when you `add_missing_bounds` it is because you got an error and now need to fix it by adding bounds for a specific axis.

That's true. I was thinking of consistency across APIs but the use cases aren't exactly the same. We can keep those APIs as is.

**Comment by pochedls** [comment]
This is currently working on 0.7.1 (even though the function signature documents an argument of form `axes : List[str]`). I'm pretty sure this did not work in the past, but it is working for me now (I should have put in a minimal example...). I think this is because if I [iterate over](https://github.com/xCDAT/xcdat/blob/main/xcdat/bounds.py#L166) `['X', 'Y']` or `X` both return a string ("X" and "Y" versus "X", respectively). We can re-open this if stops working. 

An example (that works):
`wget https://www.ncei.noaa.gov/data/noaa-global-surface-temperature/v6/access/gridded/NOAAGlobalTemp_v6.0.0_gridded_s185001_e202405_c20240608T152402.nc` (or a recent dataset [here](https://www.ncei.noaa.gov/data/noaa-global-surface-temperature/v6/access/gridded/))

```
import xcdat as xc
fn = '/p/user_pub/climate_work/pochedley1/surface/NOAAGlobalTemp_v6.0.0_gridded_s185001_e202405_c20240608T152402.nc'
ds = xc.open_dataset(fn)
ds.bounds.get_bounds('T')
```

> KeyError: "No bounds data variables were found for the 'T' axis. Make sure the dataset has bound data vars and their names match the 'bounds' attributes found on their related time coordinate variables. Alternatively, you can add bounds with `ds.bounds.add_missing_bounds()` or `ds.bounds.add_bounds()`."

```
ds = ds.bounds.add_missing_bounds('T')
ds.bounds.get_bounds('T')
```

> <xarray.DataArray 'time_bnds' (time: 2093, bnds: 2)> Size: 33kB
> array([[cftime.DatetimeGregorian(1850, 1, 1, 0, 0, 0, 0, has_year_zero=False),
>         cftime.DatetimeGregorian(1850, 2, 1, 0, 0, 0, 0, has_year_zero=False)],
> ...
>        [cftime.DatetimeGregorian(2024, 5, 1, 0, 0, 0, 0, has_year_zero=False),
>         cftime.DatetimeGregorian(2024, 6, 1, 0, 0, 0, 0, has_year_zero=False)]],
>       dtype=object)
> Coordinates:
>   * time     (time) object 17kB 1850-01-01 00:00:00 ... 2024-05-01 00:00:00
> Dimensions without coordinates: bnds
> Attributes:
>     xcdat_bounds:  True

**Comment by tomvothecoder** [comment]
> This is currently working on 0.7.1 (even though the function signature documents an argument of form `axes : List[str]`). I'm pretty sure this did not work in the past, but it is working for me now (I should have put in a minimal example...). I think this is because if I [iterate over](https://github.com/xCDAT/xcdat/blob/main/xcdat/bounds.py?rgh-link-date=2024-07-09T00%3A15%3A11Z#L166) `['X', 'Y']` or `X` both return a string ("X" and "Y" versus "X", respectively). We can re-open this if stops working.

The for loop below is **supposed to loop over a list of strings**. However, if we pass a **single string**, it'll loop over each character of the string. In your case you pass a single character string (e.g., "T") which means the for-loop behaves correctly (unintentionally). 
https://github.com/xCDAT/xcdat/blob/7751452c0f1e68c7adb51098ac4cef774769158a/xcdat/bounds.py#L166-L170

We should do the following:
1. Update the docstring with `axes: List[CFAxisKey] | CFAxisKEy`
2. Add a check to make sure the single character is a `CFAxisKey` and raise a clear `ValueError` if it isn't
